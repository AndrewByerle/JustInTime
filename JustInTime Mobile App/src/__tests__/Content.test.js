import { render, screen } from '@testing-library/react-native';
import ContentScreen from './../../src/navigation/screens/Content';

jest.useFakeTimers();

const parseDateTime = (datetime) => {
    const date = new Date('2022/11/28 18:24:43');

    const month = date.toLocaleDateString('en-us', {
        month: 'long',
    });

    const dayNum = date.getDate();

    const time = new Intl.DateTimeFormat('en-us', {
        timeStyle: 'short',
    }).format(date);

    return `${month} ${dayNum}${
        new Date().getFullYear() !== date.getFullYear()
            ? ', ' + date.getFullYear()
            : ''
    } at ${time}`;
};

test('integration check content page renders properly', () => {
    let testTitle = 'test title randomly generated';
    render(
        <ContentScreen
            route={{
                params: {
                    title: testTitle,
                    content: 'Test content',
                    datetime: new Date('2022/11/28 18:24:43'),
                },
            }}
        />
    );
    expect(screen.getByText(testTitle)).toBeTruthy();
    expect(screen.getByText('Test content')).toBeTruthy();
    expect(
        screen.getByText(parseDateTime(new Date('2022/11/28 18:24:43')))
    ).toBeTruthy();
});

test('check that back button exists', () => {
    render(
        <ContentScreen
            route={{
                params: {
                    title: 'Test title',
                    content: 'Test content',
                    datetime: new Date('2022/11/28 18:24:43'),
                },
            }}
        />
    );
    expect(screen.getByText('Back', { exact: false })).toBeTruthy();
});
