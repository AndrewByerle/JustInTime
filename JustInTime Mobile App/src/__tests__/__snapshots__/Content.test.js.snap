// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check for same content page snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
        "paddingHorizontal": 24,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        {
          "color": "#4B9CD3",
          "fontSize": 20,
        }
      }
    >
      ‚Üê Back
    </Text>
  </View>
  <RCTScrollView
    contentContainerStyle={
      {
        "paddingBottom": 50,
      }
    }
    style={
      {
        "marginBottom": 0,
        "paddingHorizontal": 24,
      }
    }
  >
    <View>
      <Text
        style={
          {
            "color": "#4B9CD3",
            "fontSize": 30,
            "fontWeight": "bold",
            "paddingVertical": 8,
          }
        }
      >
        Test title
      </Text>
      <Text
        style={
          {
            "color": "gray",
            "fontSize": 20,
            "paddingVertical": 8,
          }
        }
      >
        November 28, 6:10 PM
      </Text>
      <Text
        style={
          {
            "fontSize": 20,
            "paddingVertical": 8,
          }
        }
      >
        Test content
      </Text>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
